### PR TITLE
ARM Template To Deploy The Container App

### PR DESCRIPTION
**Pull request by QuanWanxx**
Add the ARM template to deploy an [Azure Container App](https://learn.microsoft.com/en-us/azure/container-apps/?ocid=AID3042118)

Temporarily, I retain the previous deploy container instance template if you are still use that for testing, it should be removed in OSS release.

### CODE REVIEW

File: FhirToDataLake/deploy/templates/DeployContainerApp.json [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+              ],
+              "resources": {
+                "cpu": 2,
+                "memory": "4Gi"
+              }
+            }
+          ],
+          "scale": {
+            "minReplicas": 1,
+            "maxReplicas": 3
```
Comment (moria97) [inline FhirToDataLake/deploy/templates/DeployContainerApp.json:305]: can we use a parameter like maxInstanceCount with default value 5?

Comment (QuanWanxx) [reply FhirToDataLake/deploy/templates/DeployContainerApp.json]: Sure, updated.

File: FhirToDataLake/deploy/templates/DeployContainerApp.json [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+        }
+    },
+    "containerName": {
+        "type": "string",
+        "defaultValue": "fhir",
+        "metadata": {
+            "description": "The name of the container to store job and data."
+        }
+    },
+    "jobConcurrency": {
```
Comment (hongyh13) [inline FhirToDataLake/deploy/templates/DeployContainerApp.json:56]: the concurrent sets to 10 now?

File: FhirToDataLake/deploy/templates/DeployContainerApp.json [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+                    "value": "[parameters('fhirVersion')]"
+                },
+                {
+                    "name": "fhirServer__authentication",
+                    "value": "[if(parameters('authentication'), 'ManagedIdentity', 'None')]"
+                }
+              ],
+              "resources": {
+                "cpu": 2,
+                "memory": "4Gi"
```
Comment (hongyh13) [inline FhirToDataLake/deploy/templates/DeployContainerApp.json:308]: how about making the container app's memory configurable?

**Review by moria97 — APPROVED** [review]

File: FhirToDataLake/deploy/templates/DeployContainerApp.json (hunk 2 above)
Comment (QuanWanxx) [reply FhirToDataLake/deploy/templates/DeployContainerApp.json:56]: Set concurrency to 3 pre offline discussion.

File: FhirToDataLake/deploy/templates/DeployContainerApp.json (hunk 3 above)
Comment (QuanWanxx) [reply FhirToDataLake/deploy/templates/DeployContainerApp.json:313]: Keep core as 2 and memory as 4Gi pre offline discussion.

**Pull request merged**